src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -1,39 +1,34 @@
 package eu.kanade.tachiyomi.extension.fr.scanmanga
 
 import eu.kanade.tachiyomi.network.GET
-import eu.kanade.tachiyomi.network.asObservableSuccess
 import eu.kanade.tachiyomi.source.model.FilterList
-import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import eu.kanade.tachiyomi.source.online.ParsedHttpSource
 import eu.kanade.tachiyomi.util.asJsoup
 import kotlinx.serialization.json.Json
-import kotlinx.serialization.json.jsonArray
-import kotlinx.serialization.json.jsonObject
-import kotlinx.serialization.json.jsonPrimitive
 import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.Response
 import org.jsoup.nodes.Document
 import org.jsoup.nodes.Element
-import org.jsoup.parser.Parser
-import rx.Observable
 import uy.kohesive.injekt.injectLazy
-import kotlin.random.Random
 
 class ScanManga : ParsedHttpSource() {
 
     override val name = "Scan-Manga"
 
-    override val baseUrl = "https://www.scan-manga.com"
+    override val baseUrl = "https://m.scan-manga.com"
 
     override val lang = "fr"
 
     override val supportsLatest = true
 
+    private val desktopBaseUrl = "https://www.scan-manga.com" // URL desktop pour la recherche
```

Comment:
Prefer English code comments
```suggestion
    private val desktopBaseUrl = "https://www.scan-manga.com" // Desktop URL for search
```